src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineView.xaml.cs

```diff
@@ -0,0 +1,175 @@
+
+            if (ErrorHandler.Succeeded(codeWindow.GetSecondaryView(out var secondaryTextView)))
+            {
+                if (ErrorHandler.Failed(StartTrackingView(secondaryTextView)))
+                    Debug.Fail("StartTrackingView failed during DocumentOutlineControl initialization.");
+            }
+
+            _codeWindowEventsSink = ComEventSink.Advise<IVsCodeWindowEvents>(codeWindow, this);
+        }
+
+        /// <summary>
+        /// Called by our View Model to tell us that navigation in the document has completed
+        /// </summary>
+        private void NavigationCompleted(object sender, EventArgs e)
+        {
+            _isNavigating = false;
+        }
+
+        private int StartTrackingView(IVsTextView textView)
+        {
+            var wpfTextView = _editorAdaptersFactoryService.GetWpfTextView(textView);
+            if (wpfTextView is null)
+                return VSConstants.E_FAIL;
+
+            _trackedTextViews.Add(textView, wpfTextView);
+
+            // In the split window case, there's two views (each with its own caret position) but only one text buffer.
+            // Subscribe to caret position changes once per view.
+            wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+
+            return VSConstants.S_OK;
+        }
+
+        private void ExpandAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Expand);
+
+        private void CollapseAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Collapse);
+
+        private void SearchBox_TextChanged(object sender, EventArgs e)
+        {
+            var newText = SearchBox.Text ?? string.Empty;
+            _viewModel.EnqueueFilter(newText);
+        }
+
+        private void SortByName(object sender, EventArgs e)
+            => UpdateSort(SortOption.Name, nameof(DocumentSymbolDataViewModel.Name));
+
+        private void SortByOrder(object sender, EventArgs e)
+            => UpdateSort(SortOption.Location, nameof(DocumentSymbolDataViewModel.StartPosition));
+
+        private void SortByType(object sender, EventArgs e)
+            => UpdateSort(SortOption.Type, nameof(DocumentSymbolDataViewModel.SymbolKind), nameof(DocumentSymbolDataViewModel.Name));
+
+        private void UpdateSort(SortOption sortOption, params string[] sortProperties)
+        {
+            // Log which sort option was used
+            Logger.Log(sortOption switch
+            {
+                SortOption.Name => FunctionId.DocumentOutline_SortByName,
+                SortOption.Location => FunctionId.DocumentOutline_SortByOrder,
+                SortOption.Type => FunctionId.DocumentOutline_SortByType,
+                _ => throw new NotImplementedException(),
+            }, logLevel: LogLevel.Information);
+
+            var view = ((CollectionViewSource)FindResource("DocumentSymbolItems")).View;
```

Comment:
dont' love this line.  can we somehow do this once and then refer to the strongly-typed item from taht point on?